### PR TITLE
[bugfix] Fix expected OpenMP versions for Dom

### PR DESCRIPTION
**Pull request by teojgo**
* Distinguish daint and dom openmp versions after dom maintenance.

* Refactor to new check syntax.

### CODE REVIEW

**Comment by teojgo** [comment]
@jgphpc Here are the `OpenMP` versions per PrgEnv and programming language:

- PrgEnv-cray
  - C: 201511
  - C++: 201511
  - F90: 201511
- PrgEnv-gnu
  - C: 201511
  - C++: 201511
  - F90: 201307
- PrgEnv-intel
  - C: 201611
  - C++: 201611
  - F90: 201611
- PrgEnv-pgi
  - C: 201307
  - C++: 201307
  - F90: 201307

File: cscs-checks/tools/profiling_and_debugging/notool.py [hunk 1]
```diff
@@ -81,21 +88,15 @@ def compile(self):
...
             sn.assert_eq(found_version, self.openmp_versions[ompversion_key]),
             sn.assert_found('SUCCESS', self.stdout),
         ])
-
-        self.reference['*:elapsed_time'] = self.reference_prgenv[self.current_environ.name]
-
-
-def _get_checks(**kwargs):
-    return [JacobiNoToolHybrid(lang, **kwargs) for lang in ('C', 'C++', 'F90')]
+        environ_name = self.current_environ.name
```
Comment (vkarak) [inline cscs-checks/tools/profiling_and_debugging/notool.py]: Move this up and use it also for setting the `ompversion_key`.

File: cscs-checks/tools/profiling_and_debugging/notool.py [hunk 2]
```diff
@@ -1,15 +1,17 @@
 import os
 
+import reframe as rfm
 import reframe.utility.sanity as sn
 from reframe.core.fields import ScopedDict
 from reframe.core.pipeline import RegressionTest
```
Comment (vkarak) [inline cscs-checks/tools/profiling_and_debugging/notool.py]: Remove this and access `RegressionTest` from `rfm`.

File: cscs-checks/tools/profiling_and_debugging/notool.py [hunk 3]
```diff
@@ -1,15 +1,17 @@
 import os
```
Comment (vkarak) [inline cscs-checks/tools/profiling_and_debugging/notool.py]: This is not needed.

File: cscs-checks/tools/profiling_and_debugging/notool.py [hunk 4]
```diff
@@ -1,15 +1,17 @@
...
 class JacobiNoToolHybrid(RegressionTest):
-    def __init__(self, lang, **kwargs):
-        super().__init__('jacobi_%s' % lang.replace('+', 'p'),
-                         os.path.dirname(__file__), **kwargs)
+    def __init__(self, lang):
+        super().__init__()
 
+        self.name = 'jacobi_%s' % lang.replace('+', 'p')
         self.language = lang
         self.descr = '%s check' % lang
```
Comment (vkarak) [inline cscs-checks/tools/profiling_and_debugging/notool.py]: I would put `descr` just after `name`.

**Comment by teojgo** [comment]
@vkarak once daint is upgraded we can put the newer versions to it as well.

File: cscs-checks/tools/profiling_and_debugging/notool.py (hunk 3 above)
Comment (teojgo) [reply cscs-checks/tools/profiling_and_debugging/notool.py]: It is needed for the path join according to the language.

**Comment by vkarak** [comment]
@jenkins-cscs retry dom

**Pull request merged**